doc/source/_templates/layout.html

```diff
@@ -1,6 +1,134 @@
+    '1.9.3': '1.9.3',
+    '1.8.1': '1.8.1',
+    '1.7.0': '1.7.0',
+  };
+
+  function build_version_select(current_version, current_release) {
+    var buf = ['<select>'];
+    console.log('build_version_select checkpoint A');
+
+    $.each(all_versions, function(version, title) {
+      buf.push('<option value="' + version + '"');
+      if (version == current_version)
+        buf.push(' selected="selected">' + '</option>');
+      else
+        buf.push('>' + title + '</option>');
+    });
+
+    buf.push('</select>');
+    return buf.join('');
+  }
+
+  function navigate_to_first_existing(urls) {
+    // Navigate to the first existing URL in urls.
+    console.log('navigate_to_first_existing checkpoint A');
+    var url = urls.shift();
+    if (urls.length == 0) {
+      window.location.href = url;
+      return;
+    }
+    $.ajax({
+      url: url,
+      success: function() {
+        window.location.href = url;
+      },
+      error: function() {
+        navigate_to_first_existing(urls);
+      }
+    });
+  }
+
+  function on_version_switch() {
+    console.log('on_version_switch checkpoint A');
+    var selected_version = $(this).children('option:selected').attr('value') + '/';
+    console.log('selected_version: ' + selected_version);
+    var url = window.location.href;
+    console.log('current url: ' + url);
+    var current_version = version_segment_in_url(url);
+    console.log('current_version: ' + current_version);
+    if (selected_version != 'dev/') {
+	    // selected a version that is non-local
+	    // replace the hostname with known hostname so we
+	    // can navigate from local build page to same page
+            // in different version online
+  	    var pathname = url.split('/html/')[1];
+	    var new_url = 'https://docs.scipy.org/doc/numpy-' + selected_version +  pathname;
+    }
+    else {
+	    // need to worry about local doc build here
+	    var new_url = url.replace(current_version, selected_version);
+    }
+    console.log('on_version_switch checkpoint B');
+    console.log('new_url: ' + new_url);
+    if (new_url != url) {
+      console.log('on_version_switch checkpoint C');
+      navigate_to_first_existing([
+        new_url,
+        //url.replace('VERSION', selected_version),
+        //'https://docs.scipy.org/doc/numpy-1.15.0',
+        //'https://docs.python.org/'
+      ]);
+      console.log('on_version_switch checkpoint D');
+    }
+  console.log('on_version_switch checkpoint E');
+  }
+
+  // Returns the path segment of the version as a string, like '3.6/'
+  // or '' if not found.
+  function version_segment_in_url(url) {
+    console.log('version_segment_in_url checkpoint A');
+    url = 'https://docs.scipy.org/doc/numpy-1.15.0/index.html'
+    file:///Users/treddy/github_projects/numpy_linked_worktree/doc/build/html/index.html
+    if (url.includes('file:///')) {
+        console.log('local dev doc build detected');
+	return 'dev';
+    }
+    else if (url.includes('devdocs')) {
```

Comment:
Note that this `includes` is still somewhat new, and 12% of users have browsers that don't support it: https://caniuse.com/#feat=es6-string-includes